hack/add-auto-tag.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+# Copyright The Enterprise Contract Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+AUTO_TAG=$(hack/derive-version.sh)
+
+if [ -z "$(git tag -l $AUTO_TAG)" ]; then
+  # Create a tag
+  echo "Creating tag $AUTO_TAG"
+  git tag -a -m 'Version tag added automatically after snapshot build' "$AUTO_TAG"
+
+else
+  # The tag exists already
+  # Hopefully this won't happen, but let's not break the build if it does
+  THIS_SHA=$(git rev-list -n1 --abbrev-commit HEAD)
+  TAG_SHA=$(git rev-list -n1 --abbrev-commit "$AUTO_TAG")
+
+  if [ "$TAG_SHA" = "$THIS_SHA" ]; then
+    # Tag is already what we wanted it to be. No big deal.
+    echo "Tag $AUTO_TAG exists already"
+
+  else
+    # This is more surprising. If you see this it's likely you want to do
+    # some debugging to figure out what happened and why
+    echo "Tag $AUTO_TAG exists already but on $TAG_SHA not $THIS_SHA. Skipping tag creation."
+    echo "You should try to figure out why this happened!"
+
```

Comment:
Will we ever notice this if it doesn't fail?

```suggestion
    exit 1
```

Reply:
Yeah, it's a good point. My reasoning was that I wanted to make it so it would never break the build, at least to begin with. Because a lot of the behavior is post-merge, it's hard to be 100% confident pre-merge.

How about we test it in main branch for a while, then if it looks like it's stable we can introduce a `exit 1` here.